checks/apps/pytorch/pytorch_allreduce.py

```diff
@@ -100,3 +100,71 @@ def bandwidth(self):
         return sn.extractsingle(r'\|\s*16GiB\s*\|\s*(?P<busbw>\S+)GBps\s*\|',
                                 self.stdout, tag='busbw', conv=float
         )
+
+
+@rfm.simple_test
+class PyTorchRCCLAllReduce(rfm.RunOnlyRegressionTest, ContainerEngineMixin):
+    valid_systems = ['+amdgpu']
+    valid_prog_environs = ['builtin']
+    num_nodes = variable(int, value=8)
+    sourcesdir = None
+    curated_images = [
+        'rocm/pytorch:rocm6.3.3_ubuntu24.04_py3.12_pytorch_release_2.4.0'
+    ]
+
+    image = parameter(curated_images) #+ latest_images)
+    executable = 'torchrun'
+    num_tasks_per_node = 1
+    env_vars = {
+        'NCCL_DEBUG': 'Info',
+    }
+    reference = {
+        '*': {'bandwidth': (91.04, -0.05, None, 'GB/s')}
```

Comment:
The test looks fine, but the performance is not very consistent and the test fails in performance (I have seen around 60GB/s). Not sure if we should reduce the reference or if this is expected?

Reply:
I am testing some env variables so I am leaving it for the moment